e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs

```diff
@@ -95,186 +90,131 @@ public async Task Message_TcpConnectionLossSendRecovery_MqttWs()
+                FaultInjection.DefaultFaultDelay,
+                FaultInjection.DefaultFaultDuration).ConfigureAwait(false);
         }
 
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        [ExpectedException(typeof(IotHubThrottledException))]
         public async Task Message_ThrottledConnectionLongTimeNoRecovery_Amqp()
         {
-            try
-            {
-                await SendMessageRecoveryAsync(
-                        TestDeviceType.Sasl,
-                        new IotHubClientAmqpSettings(),
-                        FaultInjection.FaultType_Throttle,
-                        FaultInjection.FaultCloseReason_Boom,
-                        FaultInjection.DefaultFaultDelay,
-                        FaultInjection.DefaultFaultDuration,
-                        FaultInjection.ShortRetryDuration)
-                    .ConfigureAwait(false);
-
-                Assert.Fail("None of the expected exceptions were thrown.");
-            }
-            catch (IotHubThrottledException) { }
-            catch (IotHubCommunicationException ex)
-            {
-                Assert.IsInstanceOfType(ex.InnerException, typeof(OperationCanceledException));
-            }
-            catch (TimeoutException) { }
+            await SendMessageRecoveryAsync(
+                TestDeviceType.Sasl,
+                new IotHubClientAmqpSettings(),
+                FaultInjection.FaultType_Throttle,
+                FaultInjection.FaultCloseReason_Boom,
+                FaultInjection.DefaultFaultDelay,
+                FaultInjection.DefaultFaultDuration,
+                FaultInjection.ShortRetryDuration).ConfigureAwait(false);
         }
 
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        [ExpectedException(typeof(IotHubThrottledException))]
         public async Task Message_ThrottledConnectionLongTimeNoRecovery_AmqpWs()
         {
-            try
-            {
-                await SendMessageRecoveryAsync(
-                        TestDeviceType.Sasl,
-                        new IotHubClientAmqpSettings(IotHubClientTransportProtocol.WebSocket),
-                        FaultInjection.FaultType_Throttle,
-                        FaultInjection.FaultCloseReason_Boom,
-                        FaultInjection.DefaultFaultDelay,
-                        FaultInjection.DefaultFaultDuration,
-                        FaultInjection.ShortRetryDuration)
-                    .ConfigureAwait(false);
-                Assert.Fail("None of the expected exceptions were thrown.");
-            }
-            catch (IotHubThrottledException) { }
-            catch (IotHubCommunicationException ex)
-            {
-                Assert.IsInstanceOfType(ex.InnerException, typeof(OperationCanceledException));
-            }
-            catch (TimeoutException) { }
-        }
-
-        [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
-        public async Task Message_ThrottledConnectionLongTimeNoRecovery_Http()
```

Comment:
We're on track to remove HTTP support for v2 device clients, so removing this test. This test also behaves differently than expected - service disconnects immediately after receiving this fault request, but the disconnection is an ungraceful I/O disconnect (and not a throttling/quota exceeded specific error code). Our [tests](https://github.com/Azure/azure-iot-sdk-csharp/blob/8e5c5bd102c23ddb94175823f219c4514831e0e2/e2e/test/iothub/messaging/MessageSendFaultInjectionTests.cs#L307-L328) also seem to accept TimeoutException and OperationCancelledException as acceptable exception types, which makes these tests non-deterministic in nature.